instrumentation/opentelemetry-instrumentation-falcon/tests/test_falcon.py

```diff
@@ -242,6 +260,88 @@ def test_traced_not_recording(self):
+                    self.assertIn(metric.name, _expected_metric_names)
+                    data_points = list(metric.data.data_points)
+                    self.assertEqual(len(data_points), 1)
+                    for point in data_points:
+                        if isinstance(point, HistogramDataPoint):
+                            self.assertEqual(point.count, 3)
+                            histogram_data_point_seen = True
+                        if isinstance(point, NumberDataPoint):
+                            number_data_point_seen = True
+                        for attr in point.attributes:
+                            self.assertIn(
+                                attr, _recommended_attrs[metric.name]
+                            )
+        self.assertTrue(number_data_point_seen and histogram_data_point_seen)
+
+    def test_falcon_metric_values(self):
+        expected_duration_attributes = {
+            "http.method": "GET",
+            "http.host": "falconframework.org",
+            "http.scheme": "http",
+            "http.flavor": "1.1",
+            "http.server_name": "falconframework.org",
+            "net.host.port": 80,
+            "http.status_code": 404,
+        }
+        expected_requests_count_attributes = {
+            "http.method": "GET",
+            "http.host": "falconframework.org",
+            "http.scheme": "http",
+            "http.flavor": "1.1",
+            "http.server_name": "falconframework.org",
+        }
+        start = default_timer()
+        self.client().simulate_get("/hello/756")
+        duration = max(round((default_timer() - start) * 1000), 0)
+        metrics_list = self.memory_metrics_reader.get_metrics_data()
+        for resource_metric in metrics_list.resource_metrics:
+            for scope_metric in resource_metric.scope_metrics:
+                for metric in scope_metric.metrics:
+                    for point in list(metric.data.data_points):
+                        if isinstance(point, HistogramDataPoint):
+                            self.assertDictEqual(
+                                expected_duration_attributes,
+                                dict(point.attributes),
+                            )
+                            self.assertEqual(point.count, 1)
+                            self.assertAlmostEqual(
+                                duration, point.sum, delta=10
+                            )
+                        if isinstance(point, NumberDataPoint):
+                            self.assertDictEqual(
+                                expected_requests_count_attributes,
+                                dict(point.attributes),
+                            )
+                            self.assertEqual(point.value, 0)
+
+    def test_metric_uninstrument(self):
+        self.client().simulate_request(method="POST", path="/hello/756")
+        FalconInstrumentor().uninstrument()
+        self.app = make_app()
```

Comment:
Let's discuss this in today's meeting

Reply:
Do we need to make changes to this uninstrument too?

Reply:
Yes, let's follow what we are doing for other instrumentation libraries. I am also fine with getting this merge if you want to address this is follow up PRs.

Reply:
Yes this would be more convenient to merge it for now with existing logic and address this change in a separate PR so that it will not get blocked for longer and we'll later on fix this test in that PR.

Reply:
My only fear is it will not be addressed later and goes backlog. Since it is a bug I am inclined to get that done before. We will be faster in terms of reviews if you think it's taking longer to get feedback.

Reply:
Sure, NP. I'll create an issue for the change and maybe pick it myself to fix it if you want.

Reply:
Added changes after fixing uninstrumentation. Please review.